PWGEM/PhotonMeson/Utils/gammaSelectionCuts.h

```diff
@@ -0,0 +1,165 @@
+    } else {
+      registry.fill(HIST("hCaloCuts_EMC"), 2, iCut);
+    }
+  }
+  return cut_return;
+}
+
+uint64_t doMinECutEMC(int iCut, uint64_t cutbit, aod::SkimEMCCluster const& cluster, HistogramRegistry& registry)
+{
+  uint64_t cut_return = 0;
+  if (cutbit & ((uint64_t)1 << (uint64_t)iCut)) {        // check if current cut should be applied
+    if (cluster.energy() > emccuts::EMC_minE.at(iCut)) { // check cut itself
+      cut_return |= (1 << iCut);                         // set bit of current cut to 1 for passing the cut
+    } else {
+      registry.fill(HIST("hCaloCuts_EMC"), 3, iCut);
+    }
+  }
+  return cut_return;
+}
+
+uint64_t doNCellCutEMC(int iCut, uint64_t cutbit, aod::SkimEMCCluster const& cluster, HistogramRegistry& registry)
+{
+  uint64_t cut_return = 0;
+  if (cutbit & ((uint64_t)1 << (uint64_t)iCut)) {             // check if current cut should be applied
+    if (cluster.nCells() >= emccuts::EMC_minNCell.at(iCut)) { // check cut itself
+      cut_return |= (1 << iCut);                              // set bit of current cut to 1 for passing the cut
+    } else {
+      registry.fill(HIST("hCaloCuts_EMC"), 4, iCut);
+    }
+  }
+  return cut_return;
+}
+
+uint64_t doTrackMatchingEMC(int iCut, uint64_t cutbit, aod::SkimEMCCluster const& cluster, aod::SkimEMCMTs const& tracks, HistogramRegistry& registry)
+{
+  uint64_t cut_return = 0;
+  double dEta, dPhi;
+  if (cutbit & ((uint64_t)1 << (uint64_t)iCut)) { // check if current cut should be applied
+    bool hasMatchedTrack_EMC = false;
+    for (const auto& track : tracks) {
+      dEta = track.tracketa() - cluster.eta();
+      dPhi = track.trackphi() - cluster.phi();
+      if ((fabs(dEta) <= emccuts::EMC_TM_Eta.at(iCut).at(0) + pow(track.trackpt() + emccuts::EMC_TM_Eta.at(iCut).at(1), emccuts::EMC_TM_Eta.at(iCut).at(2))) &&
+          (fabs(dPhi) <= emccuts::EMC_TM_Phi.at(iCut).at(0) + pow(track.trackpt() + emccuts::EMC_TM_Phi.at(iCut).at(1), emccuts::EMC_TM_Phi.at(iCut).at(2))) &&
+          cluster.energy() / track.trackp() < emccuts::EMC_Eoverp.at(iCut)) { // check cut itself
+        hasMatchedTrack_EMC = true;                                           // set bit of current cut to 1 for passing the cut
+      }
+    }
+    if (hasMatchedTrack_EMC) {
+      registry.fill(HIST("hCaloCuts_EMC"), 5, iCut);
+    } else {
+      cut_return |= (1 << iCut);
+    }
+  }
+  return cut_return;
+}
+
+uint64_t doPhotonCutsEMC(uint64_t cutbit, aod::SkimEMCCluster const& cluster, aod::SkimEMCMTs const& tracks, Preslice<o2::aod::SkimEMCMTs> perEMCClusterMT, HistogramRegistry& registry)
+{
+  uint64_t cut_return = 0;
+  for (int iCut = 0; iCut < 64; iCut++) {           // loop over max number of cut settings
+    if (cutbit & ((uint64_t)1 << (uint64_t)iCut)) { // check each cut setting if it is selected
+      registry.fill(HIST("hClusterEIn"), cluster.energy(), iCut);
```

Comment:
Could you pull this out before the loop? Only needs to be done once.. or?

Reply:
144 and 145 both need to be inside the loop

Reply:
sorry, I meant 146: ' auto tracksMatchedEMC = tracks.sliceBy(perEMCClusterMT, cluster.globalIndex());'